src/java.scripting/share/classes/javax/script/ScriptEngineManager.java

```diff
@@ -311,41 +209,50 @@ public ScriptEngine getEngineByExtension(String extension) {
      * @throws NullPointerException if mimeType is null.
      */
     public ScriptEngine getEngineByMimeType(String mimeType) {
-        if (mimeType == null) throw new NullPointerException();
-        //look for registered types first
-        Object obj;
-        if (null != (obj = mimeTypeAssociations.get(mimeType))) {
-            ScriptEngineFactory spi = (ScriptEngineFactory)obj;
-            try {
-                ScriptEngine engine = spi.getScriptEngine();
-                engine.setBindings(getBindings(), ScriptContext.GLOBAL_SCOPE);
-                return engine;
-            } catch (Exception exp) {
-                if (DEBUG) exp.printStackTrace();
-            }
-        }
+        return getEngineBy(mimeType, mimeTypeAssociations, ScriptEngineFactory::getMimeTypes);
+    }
 
-        for (ScriptEngineFactory spi : engineSpis) {
-            List<String> types = null;
-            try {
-                types = spi.getMimeTypes();
-            } catch (Exception exp) {
-                if (DEBUG) exp.printStackTrace();
-            }
-            if (types == null) continue;
-            for (String type : types) {
-                if (mimeType.equals(type)) {
-                    try {
-                        ScriptEngine engine = spi.getScriptEngine();
-                        engine.setBindings(getBindings(), ScriptContext.GLOBAL_SCOPE);
-                        return engine;
-                    } catch (Exception exp) {
-                        if (DEBUG) exp.printStackTrace();
-                    }
+    private ScriptEngine getEngineBy(String selector, Map<String, ScriptEngineFactory> associations, Function<ScriptEngineFactory, List<String>> valuesFn) {
```

Comment:
No objection to do some modernization of this code but probably best to avoid introducing overly long lines as it is impossible to see the changes with side-by-side diffs.

Reply:
Fair enough, I reformatted lines longer than 120 characters.